db/migrate/20141121200053_create_endpoints.rb

```diff
@@ -0,0 +1,13 @@
+class CreateEndpoints < ActiveRecord::Migration
+  def change
+    create_table :endpoints do |t|
+      t.string     :role
+      t.string     :ipaddress
+      t.string     :hostname
+      t.integer    :port
```

Comment:
@juliancheal Sorry, the joke is that in some places port is a string.  We had/have code that does `port.to_i` and it drives me nuts.